### PR TITLE
Refactore first half project for stop spreading props && few console issue fixes

### PR DESCRIPTION
**Pull request by AleksandrsKondratjevs**

### CODE REVIEW

File: packages/scandipwa/src/route/CartPage/CartPage.container.js [hunk 1]
```diff
@@ -207,19 +206,7 @@ export class CartPageContainer extends PureComponent {
         changeHeaderState({
             name: CART,
             title,
-            onEditClick: () => {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/CartPage/CartPage.container.js:210]: I assume we are not using edit  option on cartPage, at least I didn't saw any button or something like that on it.

File: packages/scandipwa/src/component/CartItemPrice/CartItemPrice.container.js [hunk 2]
```diff
@@ -30,17 +30,23 @@ export const mapDispatchToProps = () => ({});
 export class CartItemPriceContainer extends PureComponent {
     static propTypes = {
         getCartItemPrice: PropTypes.func.isRequired,
-        getCartItemSubPrice: PropTypes.func.isRequired
+        getCartItemSubPrice: PropTypes.func.isRequired,
+        currency_code: PropTypes.string.isRequired,
+        mix: PropTypes.object.isRequired
```
Comment (ejnshtein) [inline packages/scandipwa/src/component/CartItemPrice/CartItemPrice.container.js:35]: `PropTypes.object` is deprecated, as far as I remember, maybe we should use `PropTypes.shape` instead?

File: packages/scandipwa/src/component/CartOverlay/CartOverlay.component.js [hunk 3]
```diff
@@ -28,31 +27,25 @@ import './CartOverlay.style';
...
         currencyCode: PropTypes.string.isRequired,
         showOverlay: PropTypes.func.isRequired,
         activeOverlay: PropTypes.string.isRequired,
         hasOutOfStockProductsInCart: PropTypes.bool,
-        cartTotalSubPrice: PropTypes.number,
-        cartShippingPrice: PropTypes.number,
-        cartShippingSubPrice: PropTypes.number,
-        cartDisplaySettings: PropTypes.object.isRequired
+        cartTotalSubPrice: PropTypes.number.isRequired,
+        cartDisplaySettings: PropTypes.object.isRequired,
```
Comment (ejnshtein) [inline packages/scandipwa/src/component/CartOverlay/CartOverlay.component.js:37]: shape?

File: packages/scandipwa/src/route/Checkout/Checkout.container.js [hunk 4]
```diff
@@ -377,14 +380,62 @@ export class CheckoutContainer extends PureComponent {
...
+            isLoading,
+            orderID,
+            paymentMethods,
+            paymentTotals,
+            selectedShippingMethod,
+            shippingAddress,
+            shippingMethods,
+            selectedStoreAddress,
+            isPickInStoreMethodSelected
+        } = this.state;
```
Comment (ejnshtein) [inline packages/scandipwa/src/route/Checkout/Checkout.container.js:410]: We **MUST** do something about this.

This is an unacceptable amount of props and state properties forwarded to the component.

**Review by carinadues — APPROVED** [review]

**Review by ejnshtein — APPROVED** [review]

**Pull request merged**